lightning/src/events/mod.rs

```diff
@@ -502,7 +502,7 @@ pub enum HTLCHandlingType {
 	/// * The counterparty node modified the HTLC in transit,
 	/// * A probing attack where an intermediary node is trying to detect if we are the ultimate
 	///   recipient for a payment.
-	FailedPayment {
+	ReceiveFailed {
```

Comment:
In the context of HTLCHandlingType, this would communicate that receiving the htlc failed. Previously a reference to payment was in the name, but I think that was a layering violation?

Reply:
This represents all HTLCs that were intended for our node as the final node, but were rejected.

I think that `Payment` is a bad name here because `PaymentFailed` could refer to a payment you sent or one you received (it was unclear to me when I first encountered this struct). `Receive` is less ambiguous.

Reply:
I suppose some of the reasons have to do with the concept of a payment. Perhaps `PaymentReceiveFailed` is an option too, but agreed that putting `Receive` in there is most of the gain.